### DISCUSSION TITLE
How to debug Streamsync code?

### DISCUSSION THREAD

**Discussion by sriver** [writer/writer-framework#175] [Q&A]
Might need a little bit help on getting the debugging working for a Streamsync app (using VSCode). Created a separate python file for launching using streamsync.serve.get_asgi_app and then launching the Uvicorn server. This can be run using debugpy (VSCode debugger), but breakpoints etc. are not working. How are others doing debugging with VSCode or otherwise?

**Comment by FabienArcellier** [comment]
I have created this blueprint : https://github.com/FabienArcellier/blueprint-streamsync. You can run it in gitpod. It will show you on how using python debug on streamsync.

First, be sure using the last version of streamsync (v0.2.8). The version before it does not support python debugging.

**Reply by sriver** [reply]
Thanks again. Updated version did the trick, so the debugging in VSCode works now. Here's my launch configuration for anyone interested. Env parameters is not needed, but eliminates warning for frozen modules:

       {
            "name": "StreamSync debug",
            "type": "python",
            "request": "launch",
            "module": "streamsync.command_line",
            "justMyCode": true,
            "args": [
                "edit",
                "aiprofiler"
            ],
            "env": {
                "PYDEVD_DISABLE_FILE_VALIDATION": "1"
            }
        }
